src/main/java/skye/parser/TaskDecoder.java

```diff
@@ -0,0 +1,66 @@
+package skye.parser;
+
+import java.time.LocalDateTime;
+
+import skye.data.exception.DukeException;
+import skye.data.exception.DukeExceptionType;
+import skye.data.task.Deadline;
+import skye.data.task.Event;
+import skye.data.task.Task;
+import skye.data.task.ToDo;
+
+/**
+ * Represents a utility class that decodes a task from an encoded string
+ */
+public class TaskDecoder {
+
+    /**
+     * Decodes the encoded string representation of a task.
+     *
+     * @param line The encoded string representing a task in its save format
+     * @return Task
+     * @throws DukeException When an invalid save format is encountered
+     */
+    public Task decode(String line) throws DukeException {
```

Comment:
I like how the name of the method is very relevant and pretty much sums up the method function!